CardsCashCasino/Manager/TexasHoldEmManager.cs

```diff
@@ -804,9 +810,198 @@ private static void OnTimeoutEvent(object source, ElapsedEventArgs e)
+            /// <param name="playersActive">The positions of the players that are currently eligible for the pot</param>
+            public void AddToPot(int _currentBet, List<int> playerBets, List<int> playersActive)
+            {
+                if (playerBets.All(n => n == playerBets[0]))
+                {
+                    Pots.First(pot => pot.PotType == PotType.MAIN).Total += _currentBet * playerBets.Count();
+                }
+                else
+                {
+                    CreateSidePots(_currentBet, playerBets, playersActive);
+                }
+            }
+            /// <summary>
+            //Adds bets of players that were folded to the pot
+            /// </summary>
+            /// <param name="playersBets"> list of bets that were placed but folded due to a raise in that round</param>
+            public void AddFoldedBetsToPot(List<int> playerBets)
+            {
+                if (playerBets.Count() == 0)
+                {
+                    return;
+                }
+                else
+                {
+                    for (int bet = 0; bet < playerBets.Count(); bet++)
+                    {
+                        Pots[0].Total += playerBets[bet];
+                    }
+                }
+            }
+            /// <summary>
+            /// Creates side pots as needed at the end of a round of betting.
+            /// </summary>
+            /// <param name="_currentBet">The amount to add to the pot.</param>
+            /// <param name="playerBets">The list of bets from players still active after the round of betting.</param>
+            /// <param name="playersActive">The list of players eligible to win the main pot, all in players will be subtracted for side pot</param>
+            public void CreateSidePots(int _currentBet, List<int> playerBets, List<int> playersActive)
+            {
+                int allInBet = 0; //initiating variable that will hold all in wager
+                int numBets = playerBets.Count();
+
+                for (int bets = 0; bets < numBets; bets++)
+                {
+                    //at least 1 side pot is needed if function is called, lowest bet will be an all-in bet.
+                    if (bets == 0)
+                    {
+                        //all-in bet added to the pot, other players matched bets added to pot as well
+                        allInBet = playerBets.Min();
+                        AddToPot(allInBet, Enumerable.Repeat(allInBet, playerBets.Count()).ToList(), playersActive);
+
+                        //subtracting all-in bet value from all other bets, the all-in bet value has already been added to a pot
+                        for (int player = 0; player < playerBets.Count(); player++)
+                        {
+                            playerBets[player] -= allInBet; // Subtract the amount put into pot from each bet
+                        }
+                        _currentBet -= allInBet;
```

Comment:
does this edit need to be reflected outside of this method? if so, you need to pass it in as a reference parameter using the `ref` keyword.

Reply:
It shouldn't need to be reflected outside of the method, it's purpose will be fully served within the CreateSidePots method for splitting what portion of each bet belongs in the MAIN pot and the SIDE pot